miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/BoxRestController.java

```diff
@@ -252,8 +290,293 @@ private static BoxableId parseEntityIdentifier(String identifier) {
+      this.scannerName = scannerName;
+    }
+  }
+
+  public static class DiffMessage {
+    private BoxableDto original;
+    private BoxableDto modified;
+    private String action;
+
+    public BoxableDto getOriginal() {
+      return original;
+    }
+
+    public void setOriginal(BoxableDto original) {
+      this.original = original;
+    }
+
+    public BoxableDto getModified() {
+      return modified;
+    }
+
+    public void setModified(BoxableDto modified) {
+      this.modified = modified;
+    }
+
+    public String getAction() {
+      return action;
+    }
+
+    public void setAction(String action) {
+      this.action = action;
+    }
+  }
+
+  public static class ErrorMessage {
+    private String coordinates;
+    private String message;
+
+    public String getCoordinates() {
+      return coordinates;
+    }
+
+    public void setCoordinates(String coordinates) {
+      this.coordinates = coordinates;
+    }
+
+    public String getMessage() {
+      return message;
+    }
+
+    public void setMessage(String message) {
+      this.message = message;
+    }
+  }
+
+  public static class ScanResultsDto {
+    private List<BoxableDto> items;
+    private List<ErrorMessage> errors;
+    private List<DiffMessage> diffs;
+    private int rows;
+    private int columns;
+
+    public List<BoxableDto> getItems() {
+      return items;
+    }
+
+    public void setItems(List<BoxableDto> items) {
+      this.items = items;
+    }
+
+    public List<ErrorMessage> getErrors() {
+      return errors;
+    }
+
+    public void setErrors(List<ErrorMessage> errors) {
+      this.errors = errors;
+    }
+
+    public List<DiffMessage> getDiffs() {
+      return diffs;
+    }
+
+    public void setDiffs(List<DiffMessage> diffs) {
+      this.diffs = diffs;
+    }
+
+    public int getRows() {
+      return rows;
+    }
+
+    public void setRows(int rows) {
+      this.rows = rows;
+    }
+
+    public int getColumns() {
+      return columns;
+    }
+
+    public void setColumns(int columns) {
+      this.columns = columns;
+    }
+  }
+
+  /**
+   * Gets the Box Scanner scan results (map of box positions and barcodes)
+   * 
+   * @param boxId
+   * @param requestData
+   * @return a serialized box object containing the Boxable items
+   *         linked with each barcode at each position indicated by the scan results. Any errors are returned with a message containing the
+   *         type of
+   *         error (unable to read at certain positions; multiple items associated with a single barcode; unable to find box scanner), a
+   *         message
+   *         about the error, the positions that were successfully read (if applicable) and the positions at which the error was triggered
+   *         (if
+   *         applicable)
+   */
+  @RequestMapping(value = "/rest/box/{boxId}/scan/", method = RequestMethod.POST)
+  public @ResponseBody ScanResultsDto getBoxScan(@PathVariable(required = true) int boxId,
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 22 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)